pkg/importer/dataStream.go

```diff
@@ -202,6 +207,8 @@ func (d *DataStream) dataStreamSelector() (err error) {
 		r, err = d.s3()
 	case "http", "https":
 		r, err = d.http()
+	case "docker", "oci":
+		r, err = d.registry()
```

Comment:
unlike the other functions here like `http()`, `registry()` does a lot of stuff.  It downloads the container and extracts everything.  The bulk of the work.  Doesn't do much in `copy()` which is I believe is supposed to be the workhorse.

Reply:
Import form registry is different from other imports, so the handling is different. But as the rest it obtains a reader on VM image file for further processing such as (inspecting image file, its size validations and extracting it if it is archived ) Do you see a problem with such implementation?

Reply:
Yeah, I think it should do as little as possible.  I don't think it has to go to the network at all, actually.  The other methods only read as little of the file as necessary to construct the appropriate readers.  The whole file is not downloaded until copy().  The registry implementation should be similar for consistency sake but also, I could imagine a case where we would want to initialize a datastream but not actually move anything until later.

Reply:
Okay, I see, you have to download everything in `registry()` in order to see if the file is qcow2 so that it gets converted like an http download.  I understand now, but I'm still not sure we want to do this much work upfront

Reply:
Currently leave it as is